### PR TITLE
Feature/ios/location hashing

### PR DESCRIPTION
**Pull request by MichaelNeas**
#### Description:

Adds the native iOS component to support storing and retrieving scrypt hashes based on geohash and timestamp. Includes Realm migration with the addition of a hashes column, which is a List of Strings containing all possible hashes for the given timestamp. No migration code needs to go inside the block but the schema version must increase and there must exist a migrationBlock. https://realm.io/docs/swift/latest/#migrations

This PR brings in and slightly modifies a geohash struct found here: https://github.com/nh7a/Geohash/blob/master/Sources/Geohash/Geohash.swift

CryptoSwift implementation of Scrypt was performing too poorly.  Added Swift Package Managed wrapper above libscrypt found here: https://github.com/greymass/swift-scrypt

#### Linked issues:

Android PR with corrected comments: https://github.com/Path-Check/covid-safe-paths/pull/907
Previous iOS PR: https://github.com/Path-Check/covid-safe-paths/pull/898
Previously the thought was to bridge a saveLocation method from the react side but that brought up the issue that the RN dependency actually saves it's own SQLite DB of location data.  This required us to provide a transform the the library and implement the hashing natively on iOS and Android.

iOS component for:
#807
#806

#### How to test:

Run  ios/COVIDSafePathsTests/

### CODE REVIEW

**Review by yonatanmk — DISMISSED** [review]
I've tested this branch and I can confirm that the locations are being stored every 5 minutes with the corrected geohash-timestamp scrypt hashes

**Comment by MichaelNeas** [comment]
Before merging I would like to try to use a c library https://github.com/technion/libscrypt/tree/a402f4116245ce8677b3d9f4f87096b5ccbe26e9, which will require a little setup work from the swift side, but should prove beneficial for performance.

File: ios/Extensions/Array+Extensions.swift
Comment (tstirrat) [inline ios/Extensions/Array+Extensions.swift:11]: Can you pull in the tests from there too?

File: package.json
Comment (tstirrat) [inline package.json]: This already exists in `devDependencies`, remove

**Review by tstirrat — DISMISSED** [review]

**Review by yonatanmk — DISMISSED** [review]
I was able to get the location hashing working locally. Ready to merge once we get these tests sorted out.

**Comment by MichaelNeas** [comment]
Closing with new PR, much tidier with all tests working https://github.com/Path-Check/covid-safe-paths/pull/928

**Pull request closed without merging**
